### PR TITLE
second part completed

### PR DESCRIPTION
**Pull request by Martynice**

### CODE REVIEW

**Review by frostpv — APPROVED** [review]

File: src/main/java/com/internet/shop/service/ShoppingCartService.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import com.internet.shop.model.ShoppingCart;
+
+public interface ShoppingCartService {
+    ShoppingCart create(ShoppingCart shoppingCart);
+
+    ShoppingCart addProduct(ShoppingCart shoppingCart, Product product);
+
+    boolean deleteProduct(ShoppingCart shoppingCart, Product product);
+
+    void clear(ShoppingCart shoppingCart); //remove all products from the shoppingCart
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/service/ShoppingCartService.java]: let's remove this comment

File: src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    @Override
+    public ShoppingCart addProduct(ShoppingCart shoppingCart, Product product) {
+        shoppingCart.getProducts().add(product);
+        return shoppingCartDao.update(shoppingCart);
+    }
+
+    @Override
+    public boolean deleteProduct(ShoppingCart shoppingCart, Product product) {
+        boolean remove = shoppingCart.getProducts()
+                .removeIf(x -> x.getId().equals(product.getId()));
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java]:
```suggestion
        boolean remove = shoppingCart.getProducts().remove(product);
```

**Review by Sofasmile — COMMENTED** [review]
good job)

File: src/main/java/com/internet/shop/service/ShoppingCartService.java (hunk 1 above)
Comment (Martynice) [reply src/main/java/com/internet/shop/service/ShoppingCartService.java]: done

File: src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java (hunk 2 above)
Comment (Martynice) [reply src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java]: done

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**